### ISSUE TITLE
Transfer errors (especially missing blobs)

### ISSUE THREAD

**Issue by dmitrychastikov** [Azure/azure-storage-azcopy#984]
### Which version of the AzCopy was used? 
10.3.4

### Which platform are you using? (ex: Windows, Mac, Linux)
Windows

### What command did you run?
Job-Command copy https://prdnblu6tk3w46xclwqst0.blob.core.windows.net/meridianstreams/?se=2020-04-25t12%3A00%3A07z&sig=-REDACTED-&sp=rl&spr=https&srt=sco&ss=b&st=2020-04-25t04%3A00%3A07z&sv=2018-03-28 https://prdcloudbackup.blob.core.windows.net/2020-04-25-prdnblu6tk3w46xclwqst0?se=2020-04-25t12%3A00%3A07z&sig=-REDACTED-&sp=racwdl&spr=https&srt=sco&ss=b&st=2020-04-25t04%3A00%3A07z&sv=2018-03-28 --recursive=true --overwrite ifSourceNewer 

### What problem was encountered?
At the end of the log we can see:
2020/04/25 04:05:44 JobID=ca1ab9e8-1d51-b54b-4056-57d3194c869e, Part#=0, **TransfersDone=9229 of 9229**
2020/04/25 04:05:44 all parts of entire Job ca1ab9e8-1d51-b54b-4056-57d3194c869e successfully completed, cancelled or paused
2020/04/25 04:05:46 PERF: primary performance constraint is Unknown. States: W:  0, F:  0, S:  0, E:  0, T:  0, GRs: 256
2020/04/25 04:05:46 

Diagnostic stats:
IOPS: 156
End-to-end ms per request: 314
Network Errors: 0.00%
Server Busy: 11.37%

Job ca1ab9e8-1d51-b54b-4056-57d3194c869e summary
Elapsed Time (Minutes): 5.5951
**Total Number Of Transfers: 9229
Number of Transfers Completed: 9228**
Number of Transfers Failed: 1
Number of Transfers Skipped: 0
TotalBytesTransferred: 11426473541
**Final Job Status: CompletedWithErrors**

### How can we reproduce the problem in the simplest way?
It happens from time to time. The bigger the job the bigger the chance to see this end result.

### Have you found a mitigation/solution?
Not yet.

I do not quite understand how to interpret the log. 
Transfers done: 9229
Total Number Of Transfers: 9229
Number of Transfers Completed: 9228
Number of Transfers Failed: 1

Ok, one failed. What is that blob that failed and should I try to copy in again? 
I found that i can grep for "COPYFAILED" in the log file and this way figure out the offending blob.
In the cases that I could manage with, the CompletedWithErrors was for blobs that disappeared during the AzCopy execution. In from my standpoint these are not errors. The logic is - after AzCopy I run a program validator, it compares the source and the destination, and can find NO mismatches.  So this should not be treated and counted as errors. 
Another inconvenience is that grepping 5GB log files is not fun.
So the bottom line: 
1) blobs that were listed at start but do not exist at the moment of execution are not errors;
2) would like to ghave the list of offending blobs at the end if any, as an additional output, or an additional section at the end of the log. Maybe this is already implemented. Would appreciate any advice how to do with this. 
The problem is that we are trying to use AzCopy as a part of automated backup system and so far it requires DAYLY manual resolution.

**Comment by JohnRusk** [comment]
For your second question, would `azcopy jobs show --with-status=failed` help? (see https://docs.microsoft.com/en-us/azure/storage/common/storage-ref-azcopy-jobs-show)

cc @rishabpoh  and @zezha-msft re the first question. Let's discuss it.

**Comment by dmitrychastikov** [comment]
> For your second question, would `azcopy jobs show --with-status=failed` help? (see https://docs.microsoft.com/en-us/azure/storage/common/storage-ref-azcopy-jobs-show)
> 
> cc @rishabpoh and @zezha-msft re the first question. Let's discuss it.

I tried to use azcopy jobs show but whatever I tried I got "cannot parse the given Transfer Status failed". Which in itself sounds kinda cryptic. 
Maybe related to the fact that I schedule several instances of azcopy to run concurrently by means of setting env vars: 
```
$env:AZCOPY_LOG_LOCATION      = "c:\Backup\AzCopyData\$destContainer"+"_LOG"
$env:AZCOPY_JOB_PLAN_LOCATION = "c:\Backup\AzCopyData\$destContainer"+"_PLAN"
```
Of course I did the same before running azcopy jobs show.
```
PS C:\Backup\Scripts> & c:\backup\tools\azcopy\azcopy jobs show 4b4083ed-5d33-4d45-4e7f-1f595a3213c5

Job 4b4083ed-5d33-4d45-4e7f-1f595a3213c5 summary
Total Number Of Transfers: 9229
Number of Transfers Completed: 9228
Number of Transfers Failed: 1
Number of Transfers Skipped: 0
Percent Complete (approx): 100.0
Final Job Status: CompletedWithErrors

PS C:\Backup\Scripts> & c:\backup\tools\azcopy\azcopy jobs show 4b4083ed-5d33-4d45-4e7f-1f595a3213c5 --with-status=failed

cannot parse the given Transfer Status failed
PS C:\Backup\Scripts>
```

Here I assume that the error text should be made right English, and -with-status=failed should be fixed as well. But even if it works it's not really convenient to retrieve job name from somewhere after "AzCopy copy" is done, only to pass it to some other instance of AzCopy. Most logical would be to dump failed transfers into a file, or to output right as the part of the job execution "azcopy copy" or "azcopy sync". Optionally, e.g. --dump-failed=filepath. You can elaborate on the idea,

**Comment by JohnRusk** [comment]
Good points. BTW, I wonder if the parse error is just case sensitivity, and `--with-status=Failed` will work.

I agree that getting hold of a job ID is currently too difficult.

**Comment by dmitrychastikov** [comment]
@JohnRusk `azcopy jobs show xxxxx --with-status=Failed` did work indeed. And this can even be used to identify which blob failed. Unfortunately this does not help me with the problem of blobs deleted during the process. `azcopy jobs show xxxxx --with-status=Failed` spits out a line with `status Failed`. And there is no indication of the reason. In the log file, the same transfer has "404 The specified blob does not exist" which makes it more clear. But still, I would insist on treating blobs missing at the time of execution as non-errors. At least optionally. In real-life production environment it is not possible to have a slot of an hour or so (even a few minutes jobs have this problem actually) where blobs are not being deleted.

**Comment by JohnRusk** [comment]
I agree.  I've already marked this thread as "feature request". Thanks for the detailed and clear explanation of what we need to think about.
